core/src/main/java/org/jdbi/v3/core/transaction/LocalTransactionHandler.java

```diff
@@ -259,7 +259,9 @@ public <R, X extends Exception> R inTransaction(Handle handle,
                 handle.setTransactionIsolationLevel(level);
                 return inTransaction(handle, callback);
             } finally {
-                handle.setTransactionIsolationLevel(initial);
+                if (level != TransactionIsolationLevel.UNKNOWN) {
```

Comment:
Note: the implementation in `SerializableTransactionRunner` doesn't do this check.